arroyo-types/src/functions/hash.rs

```diff
@@ -0,0 +1,45 @@
+use base64::{engine::general_purpose::STANDARD_NO_PAD as encoder, Engine};
+use md5::{Digest, Md5};
+
+// would like to make this parameterized based on the hasher, since everything else looks the same
+// but I couldn't quite figure out what the common trait was between all of them...
```

Comment:
I thought about how to parametrize this:

```
fn hash(argument: String, hasher: ?) -> String {
    hasher.update(argument.as_bytes());
    let res = hasher.finalize();
    encoder.encode(&res)
}

pub fn md5(argument: String) -> String {
  hash(argument, Md5::new())
}

pub fn sha224(argument:String) -> String {
  hash(argument, Sha224::new())
}

etc.
```

But I couldn't quite figure out the right type for the hasher argument.  The `rust-crypto` hashing libs use something like `CoreWrapper<...>` and after messing around with some dyn/impl types, I just decided to have some repeated code for the sake of getting this PR out and gathering feedback.